### DISCUSSION TITLE
Automatic redirection error 404 pages to home?

### DISCUSSION THREAD

**Discussion by fontanil** [templaza/astroid-framework#650] [Q&A]
Hi,

Is it a way to redirect in Astroid One style settings a redirection to the home page for all 404 error pages, not on the generic 404 page?
I made changes on one of my sites (transfer RSBlog data to articles) and 119 pages have a new URL...
Thanks for your advice.

**Comment by Chacapamac** [comment]
Maybe the following can help?, I think you can use the Astroid Custom 404 settings to create the 404 page) and jump to the last —> 
    **Create a Menu Item for the Custom 404 Page (Optional):** section to show it inside you normal website. 

 **Bing Chat GPT answer**

   **Create the Custom 404 Page:**
        Design your custom 404 page using HTML and CSS. You can include helpful information, links to other pages, or even a search bar.
        Save this page with a meaningful name, such as 404.html or error-404.php.

**Upload the Custom Page to Your Joomla Installation:**
Access your Joomla site’s files via FTP or a file manager.
Navigate to the root directory (where Joomla is installed).
Upload your custom 404 page (e.g., 404.html or error-404.php) to this directory.

**Modify the .htaccess File:**
    Locate the .htaccess file in your Joomla root directory.
    Open it using a text editor.
    Add the following lines at the end of the file:

    ErrorDocument 404 /404.html

    Replace /404.html with the actual path to your custom 404 page.

**Test the Custom 404 Page:**
    Try accessing a non-existent URL on your Joomla site (e.g., https://yourdomain.com/nonexistent-page).
    You should see your custom 404 page instead of the default Joomla error page.

**Create a Menu Item for the Custom 404 Page (Optional):** _______________________________________
    If you want your custom 404 page to appear within your Joomla site’s navigation, create a **hidden men**u item for it:
        Go to Menus > Main Menu (or any other menu you prefer).
        Create a new menu item (e.g., “Page Not Found”).
        Set the menu item type to “System Links > External URL.”
        Enter the URL of your custom 404 page (e.g., /404.html).
        Save the menu item and set its access level to “Special.”

**MY NOTE:** Remember that any modules you add to that hidden menu will show up around your 404...
_______________________________________

Now, when visitors encounter a 404 error on your Joomla site, they’ll be directed to your custom 404 page. Make sure to provide clear instructions or helpful links to guide them back to relevant content! 😊

**Comment by fontanil** [comment]
Thanks Chacapamac, but I don't want to display a 404 page, only redirect to the main home page. Actually, the Astroid 404 settings in the style of Astroid One displays a picture and a message: is it a way to set some code in the texte area to open the home page?
In the standard Joomla! error.php page you can make changes using for example `header('Location: ' . Jroute::_("index.php", false));` but the file will be replaced on updates. Could we do the same in the 404 text area of the style?

**Comment by Chacapamac** [comment]
It will not be replace if you add the “error.php” to a Child Template.

**Comment by fontanil** [comment]
Yes (and it works) but I didn't want to use a child template, just use the style settings. ;)
To use a child template need to export the preset on the parent style and import it, it's not as simple as putting a bit of code in the style of the parent template.

**Comment by Chacapamac** [comment]
I see your point, but working from a Child template is way more powerful as you can really customize your website, even the core files.

> To use a child template need to export the preset on the parent style and import it, it's not as simple as putting a bit of code
This, normally take a minute with Astroid.


**Can I close the discussion?**

Good Day @fontanil

**Reply by fontanil** [reply]
Thanks Chacapamac! 
I'm still using child templates on a lot of sites. ;)
